File: Binarysearch/binary/binary_recursion.java
package Binarysearch.binary;

import java.util.*;

import javax.lang.model.util.ElementScanner6;

public class binary_recursion {
    public static int binarySearch(int[] arr, int target, int start, int end) {
        while (start <= end) {
            int mid = (start + end) / 2;
            if (arr[mid] == target)
                return mid;
            else if (arr[mid] > target)
                end = mid - 1;
            else
                start = mid + 1;
        }
        return -1;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int target = sc.nextInt();

        int[] arr = new int[n];
        for (int i = 0; i < n; i++)
            arr[i] = sc.nextInt();

        int result = binarySearch(arr, target, 0, n - 1);
        if (result == -1)
            System.out.println("원소가 존재하지 않습니다.");
        else
            System.out.println(result + 1);
    }
}

File: Binarysearch/binary/binary_recursion/binary_recursion.cpp
#include <bits/stdc++.h>
using namespace std;

int binary_search(vector<int> &array, int target, int start, int end)
{
    if (start > end)
        return -1;
    int mid = (start + end) / 2;

    if (array[mid] == target)
        return mid;
    else if (array[mid] > target)
        return binary_search(array, target, start, mid - 1);
    else
        return binary_search(array, target, mid + 1, end);
}
int n, target;
vector<int> arr;
int main()
{
    
    cin >> n >> target;

    for (int i = 0; i < n; i++)
    {
        int x;
        cin >> x;
        arr.push_back(x);
    }

    int result = binary_search(arr, target, 0, n - 1);

    if (result == -1)
        cout << "원소가 존재하지 않습니다." << endl;
    else
        cout << result + 1 << endl;
}